staging-service-shared/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/BatchPathProvider.java

```diff
@@ -109,6 +126,68 @@ public Path getTenantInprogressDirectory(final TenantId tenantId)
         return getPathForTenant(tenantId).resolve(INPROGRESS_FOLDER);
     }
 
+    public void monitorBatchProgressInFileSystem(final TenantId tenantId, final BatchId batchId, String threadID)
+    {
+        final Path pathForBatches = getTenantInprogressDirectory(tenantId);
+        final long[] size = new long[1];
+        try (Stream<Path> fileStream = Files.list(pathForBatches)) {
```

Comment:
Here we are only just going to track the progress of Batches being uploaded by a staging-service other than the current staging-service that is handling the getStatus API request.
Probably I need to rename the generic name `monitorBatchProgressInFileSystem` to something relevant. Can you coin any meaningful method name for this ?

Reply:
What was done here with the introduction of the `inProgressBatchesOfDifferentServices` variable was very damaging to the code.  It was much worse than when I originally made the comment about it streaming through the list again, and it was still iterating the list multiple times.

I've resolved this property in this commit:
[590f331c0db34ae520d5a2d8a635c62a07f8bd09] Code review: Extensive refactoring

The `getProgressReportFromDirectory()` method, which is what I renamed this method to, now takes the `Path` object, so that it doesn't need to re-find the relevant directory.